rpc/wrtc_call_queue_mongodb_test.go

```diff
@@ -177,37 +178,87 @@ func TestMongoDBWebRTCCallQueueMulti(t *testing.T) {
 
 		test.That(t, client.Database(mongodbWebRTCCallQueueDBName).Drop(context.Background()), test.ShouldBeNil)
 		answererQueue, err := NewMongoDBWebRTCCallQueue(
-			context.Background(), uuid.NewString()+"-answerer",
+			ctx, uuid.NewString()+"-answerer",
 			1, client, logger, func(hostnames []string, atTime time.Time) { activeAnswererChannelStub <- len(hostnames) }, nil)
 		test.That(t, err, test.ShouldBeNil)
 		defer answererQueue.Close()
 
 		host1 := primitive.NewObjectID().Hex()
 		host2 := primitive.NewObjectID().Hex()
+
+		var wg sync.WaitGroup
+		wg.Add(1)
 		go func() {
+			defer wg.Done()
 			// this is mimicking the caller - this sets the hostnames into the hosts array in the operatorLivenessLoop
 			_, _ = answererQueue.RecvOffer(ctx, []string{host1, host2})
 		}()
-		time.Sleep(time.Second * 2)
-
-		// val is zero when no hostnames are online, so repeat until ready to validate tests
-		foundHosts := false
-		ctx, _ = context.WithTimeout(ctx, time.Second*5)
-		for {
-			if ctx.Err() != nil || foundHosts {
-				break
-			}
+
+		// val is zero when no hostnames are online, so repeat until two hostnames are returned.
+		testutils.WaitForAssertion(t, func(tb testing.TB) {
+			tb.Helper()
 			val, ok := <-activeAnswererChannelStub
-			test.That(t, ok, test.ShouldBeTrue)
-			if val > 0 {
-				foundHosts = true
-				test.That(t, val, test.ShouldEqual, 2)
-			}
-		}
-		test.That(t, foundHosts, test.ShouldBeTrue)
+			test.That(tb, ok, test.ShouldBeTrue)
+			test.That(tb, val, test.ShouldEqual, 2)
+		})
+
+		cancel()
+		wg.Wait()
```

Comment:
updated this test cause I was gonna reuse it, but ended up going a different direction. but left my edits since I think it is strictly an improvement and less potential for flakes